app/controllers/manageBusinesses/cease/IncomeSourceEndDateController.scala

```diff
@@ -62,10 +62,10 @@ class IncomeSourceEndDateController @Inject()(val authorisedFunctions: FrontendA
     val hashedId: Option[String] = maybeIncomeSourceId.map(_.toHash.hash)
 
     (isAgent, isChange) match {
-      case (false, false) => routes.IncomeSourceEndDateController.submit(hashedId, incomeSourceType)
-      case (false, _) => routes.IncomeSourceEndDateController.submitChange(hashedId, incomeSourceType)
-      case (_, false) => routes.IncomeSourceEndDateController.submitAgent(hashedId, incomeSourceType)
-      case (_, _) => routes.IncomeSourceEndDateController.submitChangeAgent(hashedId, incomeSourceType)
+      case (false, false) => routes.IncomeSourceEndDateController.submit(hashedId, incomeSourceType, isAgent, isChange)
```

Comment:
the whole pattern matching here become redundant as you can simply call one line:
`routes.IncomeSourceEndDateController.submit(hashedId, incomeSourceType, isAgent, isChange)`

Reply:
Removed the redundant pattern matching